tracking/R3BTrackingS515.h

```diff
@@ -0,0 +1,247 @@
+            MUSIC_HITDATA,
+            LOS_HITDATA,
+            DET_MAX
+        };
+
+#define NOF_FIB_DET (DET_FI_LAST - DET_FI_FIRST + 1)
+
+        // Names of essential branches in the input tree
+        // do not change the order! add new data in the end
+        const char* fDetectorNames[DET_MAX + 1] = 
+        {
+            "Fi10Hit",  "Fi11Hit",  "Fi12Hit", "TofdHit",
+            "Mwpc0HitData", "FrsData", "MusicHitData",
+            "LosHit", 
+            NULL 
+        };
+
+        R3BEventHeader* fHeader;
+        std::vector<TClonesArray*> fDataItems;//input data 
+        TClonesArray* fTrackItems;//output data
+
+        TVector3 los_position; 
+        TVector3 mwpc_position; 
+        TVector3 f10_position; 
+        TVector3 f11_position; 
+        TVector3 f12_position; 
+        TVector3 tofd_position; 
+
+        TVector3 los_angles; 
+        TVector3 mwpc_angles; 
+        TVector3 f10_angles; 
+        TVector3 f11_angles; 
+        TVector3 f12_angles; 
+        TVector3 tofd_angles; 
+
+        R3BMDFWrapper * MDF_FlightPath;
+        R3BMDFWrapper * MDF_PoQ;
+        R3BMDFWrapper * MDF_TX0;
+        R3BMDFWrapper * MDF_TX1;
+        R3BMDFWrapper * MDF_TY0;
+        R3BMDFWrapper * MDF_TY1;
+
+        TString MDF_FlightPath_filename;
+        TString MDF_PoQ_filename;
+        TString MDF_TX0_filename;
+        TString MDF_TY0_filename;
+        TString MDF_TX1_filename;
+        TString MDF_TY1_filename;
+
+        Double_t mdf_data[8];//data container for the MDF function
+        unsigned long fNEvents; // Event counter
+        Int_t fTrigger;
+        Int_t fTpat;
+        Int_t maxevent;
+        Double_t GladCurrent;
+        Double_t GladReferenceCurrent;
+        Double_t reference_PoQ;
+        Bool_t DoAlignment;
+        Double_t tof_offset;//ns
+
+        //frs eliptic cuts:
+        Double_t FrsCutMeanZ;
+        Double_t FrsCutMeanAoZ;
+        Double_t FrsCutRadiusZ;
+        Double_t FrsCutRadiusAoZ;
+
+        //frs eliptic cuts:
+        Double_t FrsBrhoMin;
+        Double_t FrsBrhoMax;
+
+        //Z cut ranges in MUSIC set by SetMusicZMinMax()
+        Double_t MusicZMin;
+        Double_t MusicZMax;
+
+        //Cut on fiber hit energy set by SetFiberEnergyMinMax():
+        Double_t FiberEnergyMin;
+        Double_t FiberEnergyMax;
+
+        Double_t PoQ;
+        Double_t FlightPath;
+        Double_t ToF;
+        Double_t TX0;
+        Double_t TX1;
+        Double_t TY0;
+        Double_t TY1;
+        Double_t Beta;
+        Double_t Gamma;
+        Double_t mdf_AoZ;
+
+        //Essential constants
+        const Double_t SPEED_OF_LIGHT = 29.9792458;//cm/ns
+        const Double_t AMU = 0.9314940038;//GeV/c2
+
+        //Private method to fill output track data 
+        R3BTrack* AddTrackData(TVector3 mw,TVector3 poq, Double_t charge, Double_t aoz);
+
+
+    public:
+        ClassDef(R3BTrackingS515, 1)
+};
+
+double AlignmentErrorS515(const double * par);
```

Comment:
Define this as a private method around line 131, or public if you need it somewhere else.

Reply:
done